### ISSUE TITLE
Facebook Instant Articles RSS Feed

### ISSUE THREAD

**Issue by stevepessah** [akcdotorg/akc.org#231]
https://developers.facebook.com/docs/instant-articles

**Comment by kylejbertelsen** [comment]
In the Quick Start instructions (https://developers.facebook.com/docs/instant-articles/quickstart), in step 7, use the RSS option rather than the API option. Seems like it'd be the lightest lift.

**Comment by skt2** [comment]
Do we have any ref site/urls where I can see the instant articles? @spaulenoff

**Comment by aalattili** [comment]
@spaulenoff , we need more details on how/where to test this ticket

**Comment by spaulenoff** [comment]
Connect with Elie about further conversation regarding this ticket

**Comment by spaulenoff** [comment]
@skt2 Please research best practices on how to implement this and please return any findings you may come across

**Comment by skt2** [comment]
So, I have created a sample page and developed rss feed page for news articles which help upload the rss feed for news articles(as of now I have uploaded 20 articles for testing). I uploaded those articles in development environment of Facebook instant article page but these article do not have publish option available for development environment so I guess we need to test this on live page only. 

![instant_articles](https://cloud.githubusercontent.com/assets/16002526/15220735/912de638-1836-11e6-8703-5060c203c3d7.png)

Based on my research till now, I think once we upload any articles to Facebook page as an instant article and then someone share the article from site, Facebook checks if shared article is instant article or not(comparison based on URL) and if article matches, Facebook mark it as instant article by adding flash symbol at top right corner of article. -- This is what I am thinking how it should work but I haven't tested it as we need to publish some article first and its better to publish article on live page instead of test page so I stop at this point.
NOTE: This is new concept in Facebook as well as for us so there might be some unexpected challenges with this implementation so its hard to give estimate time for this.

**Comment by kylejbertelsen** [comment]
Please note: treat as a new ticket from here, down. This is focused on getting the FB IA RSS Feed correct and functional.

**Comment by skt2** [comment]
@kylejbertelsen when I uploaded RSS on dev, FB was showing error for such content:
![screen shot 2016-05-12 at 3 58 56 pm](https://cloud.githubusercontent.com/assets/16002526/15228520/892741ea-185a-11e6-995b-d17467b1aec5.png)

and this is how the content generated while image uploaded in wgywamp, as of now I am not seeing any way to avoid p tags in wgywamp but I'll keep looking into it. This is just for your attention and this might be problem for our content.

**Comment by skt2** [comment]
Time spent so far: 4h

**Comment by spaulenoff** [comment]
Please start on /news/ articles and content hub articles

**Comment by skt2** [comment]
@spaulenoff 
While creating RSS feed for instant articles, do we want to list all the news article first and then all the content articles OR we want to list all the articles based on created date doesn't if its news or content article?

**Comment by skt2** [comment]
Articles without category and with Open status:
/retro-dogs-a-visual-history-of-the-westminster-kennel-club-dog-show
/meet-the-handlers-the-humans-on-the-other-end-of-the-leads
/day-in-the-life-of-a-westminster-champion-the-post-show-media-tour
/choosing-a-champion-judges-face-a-tough-task-at-westminster
/champion-bloodlines-meet-the-breeders-of-westminster-winners
/meet-the-2016-westminster-kennel-club-dog-show-judges
/best-in-show-meet-the-westminster-champions1
/the-future-of-canine-sport-junior-handlers

Either such articles need to removed or set status to Closed or else this will create issue in sitemap and Instant Article RSS feed

**Comment by skt2** [comment]
**CMS Changes**
CMS > Design > Template Manager > Template Route Manager
Look for instant-articles, in Route field add /instant-articles.rss

**Comment by spaulenoff** [comment]
@skt2 Spoke with Elie and we want to create the RSS feed based on option two you listed above

**Comment by skt2** [comment]
@spaulenoff can you please get meta tag from Lindsay?
It look something like this: `<meta property="fb:pages" content="117352268678121" />`

**Comment by spaulenoff** [comment]
`<meta property="fb:pages" content="93474569120" />`

**Comment by skt2** [comment]
Thanks! also can you please confirm with @elieo that do we want to show ads as well? if yes then how?
Though I still need to check how that would be possible but let's confirm first.

**Comment by spaulenoff** [comment]
@skt2 Can you clarify what you mean by show ads?

**Comment by skt2** [comment]
These Ads:
![screen shot 2016-05-20 at 1 11 36 pm](https://cloud.githubusercontent.com/assets/16002526/15435786/5f34b4a2-1e8c-11e6-85d2-e91eeadf4c93.png)

**Comment by skt2** [comment]
@spaulenoff, here is RSS feed: http://develop2.akc.org/instant-articles.rss/
so on Instant Article page on Facebook, we can add
Claim URL: http://develop2.akc.org/
Development RSS Feed: http://develop2.akc.org/instant-articles.rss

**Comment by spaulenoff** [comment]
Thanks, Shailesh, Yes we wants ads to show up as well.

**Comment by skt2** [comment]
@spaulenoff @kylejbertelsen @toshan2014 Any idea if we can request ads URL from our Ads server like this?. e.g. https://www.adserver.com/adtype=x10
This is how FB Instant article works with Ads:
`<iframe src="https://www.adserver.com/ss;adtype=banner320x50" height="50" width="320"></iframe>`

**Comment by skt2** [comment]
@spaulenoff can we ask Yourbow if we can access Ads using URL like this: https://www.adserver.com/adtype=x10 ?

**Comment by spaulenoff** [comment]
Already asked. Just waiting on response from yourbow

**Comment by aalattili** [comment]
@spaulenoff ,
Intermittent issue on content hub, learn articles & news articles-Posting an FB comment showed an error. 
Actual results:
1- clicking on "login & post" button will display an eror msg (Object Invalid Value)
2- After clicking on 'OK' button on the error msg, 'Post' button will be disabled and comment won't be posted. but after refreshing the page post will display properly
please check the screenshot below.
![content hub-facebook comment-object invalid value](https://cloud.githubusercontent.com/assets/12289479/15467729/c3598760-20e8-11e6-9b2f-21a7a5a1e7db.jpg)

**Comment by skt2** [comment]
@aalattili This is not related to any of changes in this ticket though its good catch. Can you please add separate ticket for this?

**Comment by aalattili** [comment]
@skt2 , ok thanks
I have added a ticket for it (https://github.com/akcdotorg/akc.org/issues/275 )

**Comment by skt2** [comment]
Thanks @aalattili 
@spaulenoff It is better to upload 20 articles for first time as FB need to review the articles first and approve it. If we upload more articles it will take more time to get approval on articles and we won't be able to publish an article until it is approved.

**Comment by spaulenoff** [comment]
Ok @skt2 Please set that up and please update the RSS feed once this is approved.

**Comment by skt2** [comment]
Meta Needed to Claim URL on Facebook Page:  `<meta property="fb:pages" content="93474569120" />`

**Comment by skt2** [comment]
@elieo How's articles looking on Facebook page on development? Also let me know if we want to make any changes in RSS feed, Specially Kicker?
What do we want to set as Kicker?
![screen shot 2016-05-24 at 2 13 56 pm](https://cloud.githubusercontent.com/assets/16002526/15514963/ae3c5b56-21b9-11e6-8218-6404253cea75.png)

**Comment by skt2** [comment]
Removed Kicker as its optional, until we have alternate for this.

**Comment by skt2** [comment]
Overall Time Spent: 20h - R&D and RSS Feed development.

**Comment by spaulenoff** [comment]
@skt2 Do we have an RSS feed to check?

**Comment by spaulenoff** [comment]
or any links for that matter?

**Comment by skt2** [comment]
http://stage-ng.akc.org/instant-articles.rss/

**Comment by skt2** [comment]
http://www.akc.org/instant-articles.rss/
